python/dgl/nn/pytorch/graph_transformer.py

```diff
@@ -216,6 +240,138 @@ def forward(self, ndata, attn_bias=None, attn_mask=None):
+    Graphs <https://arxiv.org/pdf/2012.09699>`__
+
+    .. math::
+
+        \text{Attn}_{ij}=\text{softmax}_{j\in\mathcal{N}(i)}(\dfrac{Q_iK_j^T}
+        {\sqrt{d}} \circ b)
+
+    :math:`Q_i` and :math:`K_j` are feature representation of nodes. :math:`d`
+    is the corresponding :attr:`feat_size`. :math:`b` is attention bias, which
+    can be additive or multiplicative according to the operator :math:`\circ`.
+    :math:`\mathcal{N}(i)` is the set of source nodes that have an edge to the
+    destination node :math:`i`.
+
+    Parameters
+    ----------
+    feat_size : int
+        Feature size.
+    num_heads : int
+        Number of attention heads, by which attr:`feat_size` is divisible.
+    bias : bool, optional
+        If True, it uses bias for linear projection. Default: True.
+    attn_bias_type : str, optional
+        The type of attention bias used for modifying attention. Selected from
+        'add' or 'mul'. Default: 'add'.
+
+        * 'add' is for additive attention bias.
+        * 'mul' is for multiplicative attention bias.
+    attn_drop : float, optional
+        Dropout probability on attention weights. Defalt: 0.
+
+    Examples
+    --------
+    >>> import dgl
+    >>> import torch as th
+    >>> from dgl.nn import SparseBiasedMultiheadAttention
+
+    >>> g = dgl.rand_graph(100, 80)
+    >>> ndata = th.rand(100, 512)
+    >>> bias = th.rand(80, 8)
+    >>> net = SparseBiasedMultiheadAttention(feat_size=512, num_heads=8)
+    >>> out = net(g, ndata, bias)
+    >>> print(out.size())
+    torch.Size([100, 512])
+    """
+
+    def __init__(
+        self, feat_size, num_heads, bias=True, attn_bias_type='add',
+        attn_drop=0.
+    ):
+        super().__init__()
+        self.feat_size = feat_size
+        self.num_heads = num_heads
+        self.head_dim = feat_size // num_heads
+        assert (
+            self.head_dim * num_heads == feat_size
+        ), "feat_size must be divisible by num_heads"
+        self.scaling = self.head_dim**-0.5
+        self.attn_bias_type = attn_bias_type
+
+        self.q_proj = nn.Linear(feat_size, feat_size, bias=bias)
+        self.k_proj = nn.Linear(feat_size, feat_size, bias=bias)
+        self.v_proj = nn.Linear(feat_size, feat_size, bias=bias)
+        self.out_proj = nn.Linear(feat_size, feat_size, bias=bias)
+
+        self.dropout = nn.Dropout(p=attn_drop)
+
+        self.reset_parameters()
+
+    def reset_parameters(self):
+        """Reset parameters of projection matrices, the same settings as that
+        in Graphormer.
+        """
+        nn.init.xavier_uniform_(self.q_proj.weight, gain=2**-0.5)
+        nn.init.xavier_uniform_(self.k_proj.weight, gain=2**-0.5)
+        nn.init.xavier_uniform_(self.v_proj.weight, gain=2**-0.5)
+
+        nn.init.xavier_uniform_(self.out_proj.weight)
+        if self.out_proj.bias is not None:
+            nn.init.constant_(self.out_proj.bias, 0.0)
+
+    def forward(self, g, ndata, attn_bias=None):

```

Comment:
Why not use `feat` or `nfeat` as in most existing NN modules? Same for `BiasedMultiheadAttention`